woocommerce/woocommerce
# Add PR links to release post template output, and allow blog ID to be specified.

### All Submissions:






### Changes proposed in this Pull Request:

This PR updates the release post tool such that it's possible to edit previous release posts. It also adds support for specifying a blog ID, if you'd like to test the tool with a different blog. Finally, it links to the PRs that introduced the changes in the templates section.



Closes #34248.






### How to test the changes in this Pull Request:



1. Make sure Docker is running and in a state to run `wp-env`
1. Checkout this branch
1. `cd` to the `tools/release-posts` directory
1. Follow the [instructions in the readme](https://github.com/woocommerce/woocommerce/blob/trunk/tools/release-posts/README.md#publishing-draft-posts) to setup for publishing release posts. If you're doing this with a test blog, use that site ID when granting access.
1. The tool will open a browser window, so be sure before running that your focused browser is logged in with access to the blog you'll use for testing.
1. Run the following: `GITHUB_ACCESS_TOKEN="..." WPCOM_OAUTH_CLIENT_ID="..." WPCOM_OAUTH_CLIENT_SECRET="..." pnpm run release-post 7.2.0 7.1.0 --siteId=...`  Be sure to provide th values for the missing variables (...). You may also choose to specify the values via your `.env` file.
1. Follow the prompts in the open browser tool to grant permissions to the site corresponding to the `siteId` that you specified.
1. Verify that the post that is created contains a link to the PR(s) that introduced the template changes.
1. Create a new post on your test site manually, add ` This will be replaced ` to the source of the post you created. Note the post id.
1. Re-run the previous command with `--editPost=...`, specifying the ID of the post you just created.
1. Verify that that section of the post gets updated properly.



### Other information:




### FOR PR REVIEWER ONLY:
